src/functions/vacation-request/list-vacation-request/handler.ts

```diff
@@ -6,15 +6,27 @@ import { middyfy } from "src/libs/lambda";
 /**
  * Labmda for listing all vacation requests from DynamoDB.
  */
-const listVacationRequestHandler: APIGatewayProxyHandler = async () => {
+const listVacationRequestHandler: APIGatewayProxyHandler = async (event) => {
+  const userId = event.queryStringParameters?.userId;
   try {
-    const allVacationRequests: VacationRequestModel[] =
-      await vacationRequestService.listVacationRequests();
+    if (userId) {
```

Comment:
What if !userIf ? Do we need to consider that ?

Reply:
If it`s !userId then we skip this condition and make a request for all vacation requests on the next lines